hail/src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -127,14 +127,25 @@ abstract class AbstractMatrixTableSpec extends RelationalSpec {
     val entries = AbstractRVDSpec.read(HailContext.get, path + "/" + entriesComponent.rel_path)
     RVDType(rows.encodedType.appendKey(MatrixType.entriesIdentifier, entries.encodedType), rows.key)
   }
+
+  def rowsTableSpec(path: String): AbstractTableSpec = RelationalSpec.read(HailContext.get, path).asInstanceOf[AbstractTableSpec]
+  def colsTableSpec(path: String): AbstractTableSpec = RelationalSpec.read(HailContext.get, path).asInstanceOf[AbstractTableSpec]
+  def entriesTableSpec(path: String): AbstractTableSpec = RelationalSpec.read(HailContext.get, path).asInstanceOf[AbstractTableSpec]
 }
 
 case class MatrixTableSpec(
   file_version: Int,
   hail_version: String,
   references_rel_path: String,
   matrix_type: MatrixType,
-  components: Map[String, ComponentSpec]) extends AbstractMatrixTableSpec
+  components: Map[String, ComponentSpec]) extends AbstractMatrixTableSpec {
+
+  // some legacy files written as MatrixTableSpec wrote the wrong type to the entries table metadata
+  override def entriesTableSpec(path: String): AbstractTableSpec = {
+    val writtenETS = super.entriesTableSpec(path).asInstanceOf[TableSpec]
+    writtenETS.copy(table_type = TableType(matrix_type.entriesRVType, FastIndexedSeq(), matrix_type.globalType))
```

Comment:
the entries table spec uses the MatrixTable global type? Aren't they stored separately?

Reply:
as discussed, the globals are a relative path.